web/src/beta/components/ListItem/index.tsx

```diff
@@ -42,6 +46,10 @@ const ListItem: FC<Props> = ({
           children
         )}
       </Inner>
+
+      <HideLayer onClick={onLayerUpdate} isSelected={isSelected}>
+        {value}
+      </HideLayer>
```

Comment:
Is there any problem with this being passed inside the children node? That way you don't have to change this ListItem at all..